compiler/q-implant/src/QImplant.cpp

```diff
@@ -251,6 +271,50 @@ void QImplant::write(loco::Graph *g)
     }
   }
 
+  std::set<luci::CircleNode *> visit;
+  std::deque<luci::CircleNode *> que;
+  for (auto input_node : loco::input_nodes(g))
+  {
+    auto node = loco::must_cast<luci::CircleNode *>(input_node);
+    que.emplace_back(node);
+  }
+
+  // check whether node can share its successor node quantization parameter
+  while (!que.empty())
+  {
+    auto node = que.front();
+    que.pop_front();
+
+    // skip when node is output
+    if (node->opcode() == luci::CircleOpcode::CIRCLEOUTPUT)
+    {
+      continue;
+    }
+
+    auto quantparam = node->quantparam();
+    THROW_UNLESS(quantparam);
+
+    for (auto child : loco::succs(node))
+    {
+      auto child_node = loco::must_cast<luci::CircleNode *>(child);
+      if (visit.find(child_node) != visit.end())
+        continue;
+      visit.emplace(child_node);
+      que.emplace_back(child_node);
+
+      if (child_node->quantparam() == nullptr)
+      {
+        if (appendable_operator_opcode.find(child_node->opcode()) ==
+            appendable_operator_opcode.end())
+          continue;
+
+        set_dtype(node, child_node);
+        child_node->quantparam(std::make_unique<luci::CircleQuantParam>());
```

Comment:
Better to move this inside `copy_quantparam`.

Reply:
For using `copy_quantparam` from `CircleQuantParam`, just put `copy_dtype` as before.